pkg/hubble/filters/tracing.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package filters
+
+import (
+	"context"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func filterByTraceID(tids []string) FilterFunc {
+	return func(ev *v1.Event) bool {
+		traceID := ev.GetFlow().GetTraceContext().GetParent().GetTraceId()
+		if traceID == "" {
+			return false
+		}
```

Comment:
As it stand, as soon as there is a `trace-id` filter then all flows without `trace-id` are going to be filtered out (regardless of the filter's value). Without this branch, a filter with an empty string `trace-id` (such as in the "empty filter" test case) could match flows without a `trace-id`, thus allowing to express a "all flows without `trace-id`" filter (and "all flows with `trace-id`" through the exclusion filter). Does that make sense?
```suggestion
```

Reply:
Good call, I've remove this check and added 2 test cases to validate this behavior.